chemprop/nn/loss.py

```diff
@@ -182,41 +181,48 @@ def _calc_unreduced_loss(self, preds: Tensor, targets: Tensor, *args) -> Tensor:
         return F.cross_entropy(preds, targets, reduction="none")
 
 
-class MccMixin:
-    """Calculate a soft Matthews correlation coefficient ([mccWiki]_) loss for multiclass
-    classification based on the implementataion of [mccSklearn]_
-
-    References
-    ----------
-    .. [mccWiki] https://en.wikipedia.org/wiki/Phi_coefficient#Multiclass_case
-    .. [mccSklearn] https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html
-    """
-
-    def __call__(self, preds: Tensor, targets: Tensor, mask: Tensor, weights: Tensor, *args):
+@LossFunctionRegistry.register("binary-mcc")
+class BinaryMCCLoss(LossFunction):
+    def forward(self, preds: Tensor, targets: Tensor, mask: Tensor, weights: Tensor, *args):
         if not (0 <= preds.min() and preds.max() <= 1):  # assume logits
-            preds = preds.softmax(2)
+            preds = preds.sigmoid()
```

Comment:
Why do we change this here?

Reply:
Because this is for binary classification, the activation function should be sigmoid. This bug was identified in #846.